toqito/state_opt/tests/test_state_distinguishability.py

```diff
@@ -1,92 +1,34 @@
-
-    res = state_distinguishability(states, probs, dist_method="unambiguous")
-    np.testing.assert_equal(np.isclose(res, 1), True)
-
-
-def test_state_distinguishability_three_state_vec():
-    """State distinguishability for two state vectors."""
-    e_0, e_1 = basis(2, 0), basis(2, 1)
-    states = [e_0, e_1]
-    probs = [1 / 2, 1 / 2]
-
-    res = state_distinguishability(states, probs)
-    np.testing.assert_equal(np.isclose(res, 1), True)
-
-
-def test_state_distinguishability_yyd_density_matrices():
-    """Global distinguishability of the YYD states should yield 1."""
-    psi0 = bell(0) * bell(0).conj().T
-    psi1 = bell(1) * bell(1).conj().T
-    psi2 = bell(2) * bell(2).conj().T
-    psi3 = bell(3) * bell(3).conj().T
-
-    states = [
-        np.kron(psi0, psi0),
-        np.kron(psi2, psi1),
-        np.kron(psi3, psi1),
-        np.kron(psi1, psi1),
-    ]
-    probs = [1 / 4, 1 / 4, 1 / 4, 1 / 4]
-
-    res = state_distinguishability(states, probs)
-    np.testing.assert_equal(np.isclose(res, 1, atol=0.001), True)
-
-
-def test_invalid_state_distinguishability_probs():
-    """Invalid probability vector for state distinguishability."""
-    with np.testing.assert_raises(ValueError):
-        rho1 = bell(0) * bell(0).conj().T
-        rho2 = bell(1) * bell(1).conj().T
-        states = [rho1, rho2]
-        state_distinguishability(states, [1, 2, 3])
-
-
-def test_invalid_state_distinguishability_states():
-    """Invalid number of states for state distinguishability."""
-    with np.testing.assert_raises(ValueError):
-        states = []
-        state_distinguishability(states)
+from toqito.states import bell
+
+e_0, e_1 = standard_basis(2)
+
+
+@pytest.mark.parametrize("vectors, probs, solver, primal_dual, expected_result", [
+    # Bell states (default uniform probs with primal).
+    ([bell(0), bell(1), bell(2), bell(3)], None, "cvxopt", "primal", 1),
+    # Bell states (default uniform probs with dual).
+    ([bell(0), bell(1), bell(2), bell(3)], None, "cvxopt", "dual", 1),
+    # Bell states uniform probs with primal.
+    ([bell(0), bell(1), bell(2), bell(3)], [1/4, 1/4, 1/4, 1/4], "cvxopt", "primal", 1),
+    # Bell states uniform probs with dual.
+    ([bell(0), bell(1), bell(2), bell(3)], [1/4, 1/4, 1/4, 1/4], "cvxopt", "dual", 1),
+])
+def test_state_distinguishability(vectors, probs, solver, primal_dual, expected_result):
+    """Test function works as expected for a valid input."""
+    val, _ = state_distinguishability(vectors=vectors, probs=probs, solver=solver, primal_dual=primal_dual)
+    assert abs(val - expected_result) <=1e-8
+
+
+@pytest.mark.parametrize("vectors, probs, solver, primal_dual", [
+    # Bell states (default uniform probs with dual).
+    ([bell(0), bell(1), bell(2), e_0], None, "cvxopt", "dual"),
+])
+def test_state_distinguishability_invalid_vectors(vectors, probs, solver, primal_dual):
+    """Test function works as expected for an invalid input."""
+    with pytest.raises(ValueError):
```

Comment:
```suggestion
    with pytest.raises(ValueError, match = "Vectors for state distinguishability must all have the same dimension."):
```
It's better to use `match` because I have noticed instances of pytest thinking a line is uncovered because we did not make it check the error message.